app/src/main/java/org/dharmaseed/android/NavigationActivity.java

```diff
@@ -253,6 +289,9 @@ void setViewMode(int viewMode, boolean setMenuCheck) {
         this.viewMode = viewMode;
         header.setVisibility(View.GONE);
         extraSearchTerms = "";
+        clearSearch(false);
```

Comment:
Thanks for doing this! I agree it didn't make much sense to carry over the search state between view modes.